lib/ice_cube/validations/yearly_by_set_pos.rb

```diff
@@ -0,0 +1,77 @@
+module IceCube
+
+  module Validations::YearlyBySetPos
+
+    def by_set_pos(*by_set_pos)
+      by_set_pos.flatten!
+      by_set_pos.each do |set_pos|
+        unless (-366..366).include?(set_pos) && set_pos != 0
+          raise ArgumentError, "Expecting number in [-366, -1] or [1, 366], got #{set_pos} (#{by_set_pos})"
+        end
+      end
+
+      @by_set_pos = by_set_pos
+      replace_validations_for(:by_set_pos, [Validation.new(by_set_pos, self)])
+      self
+    end
+
+    class Validation
+
+      attr_reader :rule, :by_set_pos
+
+      def initialize(by_set_pos, rule)
+
+        @by_set_pos = by_set_pos
+        @rule = rule
+      end
+
+      def type
+        :day
+      end
+
+      def dst_adjust?
+        true
+      end
+
+      def validate(step_time, start_time)
+        start_of_year = step_time.beginning_of_year
```

Comment:
Ditto my comment from the previous file regarding `beginning_of_month`. The `beginning_of_year` method is only available via active support (defined [here](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/date_and_time/calculations.rb#L179)).

Reply:
Without active support:

```ruby
start_of_month = IceCube::TimeUtil.build_in_zone([step_time.year, 1, 1, 0, 0, 0], step_time)
```